app/src/main/java/de/westnordost/streetcomplete/quests/max_weight/Measure.kt

```diff
@@ -0,0 +1,32 @@
+package de.westnordost.streetcomplete.quests.max_weight
+
+//if the minimum required API would be 24, we could maybe use https://developer.android.com/reference/android/icu/util/Measure
+abstract class Measure {
+    abstract fun toTons(): Double
+    abstract override fun toString(): String
+}
+
+data class MetricMeasure(val tons: Double) : Measure() {
+    override fun toTons() = tons
+    override fun toString() =
+        if (tons % 1 == 0.0) {
+            tons.toInt().toString()
+        } else {
+            tons.toString()
+        }
```

Comment:
Why the if-else?

Reply:
I admit that it was copied from max height quest, without testing is it actually needed - https://github.com/westnordost/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/quests/max_height/Measure.kt#L12-L16